src/SparkFun_AS7331.h

```diff
@@ -14,1312 +14,716 @@
+
+    /// @brief Called when changing values that affect the conversion, calculates
+    /// a new conversion factor to reduce the conversion overhead.
+    void calculateConversionFactors(void);
+
+    /// @brief Called to reset all local values back to initial conditions.
+    void setDefaultSettings(void);
+
+    sfeTkIBus *_theBus;  // Pointer to bus device.
+    uint8_t _devAddress; // Device's I2C address.
+
+    uint8_t _breakTime; // Local config value. Value is in us/8. EX: _breakTime = 20 means 20*8 = 160us.
+    uint8_t _numEdges;  // Local config value. Edges seen on SYN pin before ending conversion in SYND mode.
+
+    bool _readyPinMode;         // Local config value. False is Push/Pull True is Open/Drain.
+    bool _dividerEnabled;       // Local config value. False is disabled, True is enabled.
+    bool _tempConvEnabled;      // Local config value. False is disabled, True is enabled.
+    bool _indexMode;            // Local config value. False is for controllers without repeat start.
+    bool _standbyState;         // Local state value. False means the device is not in standby mode.
+    bool _startState;           // Local state value. False means the device is not allowed to measure.
+    bool _powerDownEnableState; // Local state value. False means the device is NOT POWERED DOWN. This is inverse.
+
+    as7331_dev_op_state_t _opMode; // Local state value. Configuration or Measure operating mode.
+    as7331_gain_t _sensorGain;     // Local config value. Sensor gain stored as (1 << (11 - gain)).
+    as7331_conv_clk_freq_t _cclk;  // Local config value. Sensor's conversion clock stored as 1024*(1 << cclk) kHz.
+    as7331_meas_mode_t _mmode; // Local state value. Details the device's measurement mode as CONT, CMD, SYNS, or SYND.
+    as7331_conv_time_t _conversionTime; // Local config value. Contains the conversion time stored as (1 << time) ms.
+    as7331_divider_val_t _dividerRange; // Local config value. Contains the predivider value stored as (1 << 1 + range).
+
+    float _uva;         // Last valid UVA result in uW/cm2.
+    float _uvb;         // Last valid UVB result in uW/cm2.
+    float _uvc;         // Last valid UVC result in uW/cm2.
+    float _temperature; // Last valid temperature result in degC.
+
+    uint32_t _outputConversionTime; // Last valid output conversion time result in of clock cycle count.
+
+    float _conversionA; // Internal conversion factor for the UVA sensor.
+    float _conversionB; // Internal conversion factor for the UVB sensor.
+    float _conversionC; // Internal conversion factor for the UVC sensor.
+
+    const float _fsrA = 348160.0; // Full Scale Resolution for the UVA sensor.
+    const float _fsrB = 387072.0; // Full Scale Resolution for the UVB sensor.
+    const float _fsrC = 169984.0; // Full Scale Resolution for the UVC sensor.
+};
 
-            sfe_as7331_reg_cfg_optreg_t optreg;
+class SfeAS7331ArdI2C : public SfeAS7331Driver
```

Comment:
We probably need to standardize the naming pattern of the Arduino driver implementations - I know they are all over the place right now.  This is fine though - but something for our future selfs to hash a bit (I'm not sure if Arduino has any style guidelines here - ...).